packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -359,7 +361,28 @@ export async function collectFiles(
         return prefix === '' ? path : `${prefix}/${path}`
     }
 
-    let totalSizeBytes = 0
+    // Add target file first to the collect files if provided
+    if (filePath) {
+        const targetUri = vscode.Uri.file(filePath)
+        const relativePath = getWorkspaceRelativePath(filePath, { workspaceFolders })
+
+        if (relativePath) {
+            const fileStat = await fs.stat(targetUri)
+
+            const fileContent = await readFile(targetUri)
+            if (fileContent !== undefined) {
+                totalSizeBytes += fileStat.size
+                storage.push({
+                    workspaceFolder: relativePath.workspaceFolder,
+                    relativeFilePath: relativePath.relativePath,
+                    fileUri: targetUri,
+                    fileContent: fileContent,
+                    zipFilePath: prefixWithFolderPrefix(relativePath.workspaceFolder, relativePath.relativePath),
+                })
+            }
+        }
```

Comment:
nit: can probably simplify this since you do the same thing in lines 402-429

Reply:
Yes, please de-dup this code